--- application.py
+++ application.py_new
@@ -38,6 +38,8 @@ from keystone.oauth1 import routers as oauth1_routers
 from keystone.policy import routers as policy_routers
 from keystone.resource import routers as resource_routers
 from keystone.token import _simple_cert as simple_cert_ext
+from keystone.contrib.scim import routers as scim_routers
+from keystone.contrib.spassword import routers as spassword_routers
 
 # TODO(morgan): _MOVED_API_PREFIXES to be removed when the legacy dispatch
 # support is removed.
@@ -58,6 +60,8 @@ ALL_API_ROUTERS = [auth_routers,
                    endpoint_policy_routers,
                    ec2_routers,
                    s3_routers,
+                   scim_routers,
+                   spassword_routers,
                    # TODO(morganfainberg): Remove the simple_cert router
                    simple_cert_ext]
 
