--- token.py
+++ token.py_new
@@ -119,6 +119,10 @@ def token_authenticate(request, token):
         response_data['audit_id'] = token_audit_id
         response_data.setdefault('user_id', token.user_id)
 
+        #response_data.setdefault('extras', token.extras)
+        response_data.setdefault('extras', {}).update(
+            token.get('extras', {}))
+        LOG.debug('response_data: %s' % response_data)
         return response_data
 
     except AssertionError as e:
