--- backends.py
+++ backends.py_new
@@ -31,6 +31,7 @@ from keystone import resource
 from keystone import revoke
 from keystone import token
 from keystone import trust
+from keystone.contrib import spassword
 
 LOG = log.getLogger(__name__)
 
@@ -54,8 +55,8 @@ def load_backends():
                 identity.Manager, identity.ShadowUsersManager,
                 limit.Manager, oauth1.Manager, policy.Manager,
                 resource.Manager, revoke.Manager, assignment.RoleManager,
                 receipt.provider.Manager, trust.Manager,
-                token.provider.Manager]
+                token.provider.Manager, spassword.SPasswordManager]
 
     drivers = {d._provides_api: d() for d in managers}
 
