--- /usr/lib/python2.7/site-packages/keystone/token/providers/common.py	2017-11-03 12:23:48.935963698 +0000
+++ /usr/lib/python2.7/site-packages/keystone/token/providers/common.py_new	2017-11-03 12:30:33.426329981 +0000
@@ -498,11 +498,13 @@
             LOG.error(msg)
             raise exception.UnexpectedError(msg)
 
-    def get_token_data(self, user_id, method_names, domain_id=None,
+    def get_token_data(self, user_id, method_names, extras=None, domain_id=None,
                        project_id=None, expires=None, trust=None, token=None,
                        include_catalog=True, bind=None, access_token=None,
                        issued_at=None, audit_info=None):
-        token_data = {'methods': method_names}
+	 if extras is None:
+            extras = {}
+        token_data = {'methods': method_names, 'extras': extras}
 
         # We've probably already written these to the token
         if token:
@@ -610,6 +612,7 @@
         token_data = self.v3_token_data_helper.get_token_data(
             user_id,
             method_names,
+            auth_context.get('extras') if auth_context else None,
             domain_id=domain_id,
             project_id=project_id,
             expires=expires_at,
