--- common.py	2017-11-03 12:23:48.935963698 +0000
+++ common.py_new	2017-11-03 12:30:33.426329981 +0000
@@ -498,12 +498,14 @@
            token_data['application_credential']['name'] = app_cred['name']
            token_data['application_credential']['restricted'] = restricted
 
-    def get_token_data(self, user_id, method_names, system=None,
+    def get_token_data(self, user_id, method_names, system=None, extras=None,
                        domain_id=None, project_id=None, expires=None,
                        app_cred_id=None, trust=None, token=None,
                        include_catalog=True, bind=None, access_token=None,
                        issued_at=None, audit_info=None):
-        token_data = {'methods': method_names}
+	 if extras is None:
+            extras = {}
+        token_data = {'methods': method_names, 'extras': extras}
 
         # We've probably already written these to the token
         if token:
@@ -591,6 +593,7 @@
             user_id,
             method_names,
             system=system,
+            auth_context.get('extras') if auth_context else None,
             domain_id=domain_id,
             project_id=project_id,
             expires=expires_at,
