--- core.py   2019-01-29 13:25:28.475608664 +0000
+++ core.py_new 2019-01-29 13:25:02.063686461 +0000
@@ -832,6 +832,6 @@
         ref = driver.authenticate(entity_id, password)
         ref = self._set_domain_id_and_mapping(
             ref, domain_id, driver, mapping.EntityType.USER)
-        ref = self._shadow_nonlocal_user(ref)
+        #ref = self._shadow_nonlocal_user(ref)
         self.shadow_users_api.set_last_active_at(ref['id'])
         return ref
