--- 073_expand_initial_migration.py
+++ 073_expand_initial_migration.py_new
@@ -533,6 +533,26 @@
         mysql_charset='utf8',
     )
 
+    spassword = sql.Table(
+        'spassword',
+        meta,        
+        sql.Column('user_id', sql.String(64), primary_key=True),
+        sql.Column('user_name', sql.String(255)),
+        sql.Column('domain_id', sql.String(64)),
+        sql.Column('creation_time', sql.DateTime()),
+        sql.Column('login_attempts', sql.Integer),
+        sql.Column('last_login_attempt_time', sql.DateTime()),
+        sql.Column('extra', sql.Text()),
+        sql.Column('sndfa', sql.Boolean, default=False),
+        sql.Column('sndfa_last', sql.DateTime(), default=None),
+        sql.Column('sndfa_code', sql.String(32), default=None),
+        sql.Column('sndfa_time_code', sql.DateTime(), default=None),
+        sql.Column('sndfa_email', sql.Boolean, default=False),
+        sql.Column('sndfa_email_code', sql.String(32), default=None),
+        mysql_engine='InnoDB',
+        mysql_charset='utf8',
+    )
+
     system_assignment = sql.Table(
         'system_assignment',
         meta,
@@ -1045,6 +1065,7 @@
         sensitive_config,
         service,
         service_provider,
+        spassword,
         system_assignment,
         token,
         trust,
