--- manager.py   2020-03-04 13:25:28.475608664 +0000
+++ manager.py_new 2020-03-04 13:25:02.063686461 +0000
@@ -18,5 +18,7 @@ import time
 import types
 
 from oslo_log import log
+from oslo_log import versionutils
+from oslo_utils import importutils
 import six
 import stevedore
@@ -72,9 +74,20 @@ def load_driver(namespace, driver_name, *args):
                                                  invoke_on_load=True,
                                                  invoke_args=args)
         return driver_manager.driver
-    except stevedore.exception.NoMatches:
-        msg = (_('Unable to find %(name)r driver in %(namespace)r.'))
-        raise ImportError(msg % {'name': driver_name, 'namespace': namespace})
+    except RuntimeError as e:
+        LOG.debug('Failed to load %r using stevedore: %s', driver_name, e)
+        # Ignore failure and continue on.
+
+    driver = importutils.import_object(driver_name, *args)
+
+    msg = (_(
+        'Direct import of driver %(name)r is deprecated as of Liberty in '
+        'favor of its entrypoint from %(namespace)r and may be removed in '
+        'N.') %
+        {'name': driver_name, 'namespace': namespace})
+    #versionutils.report_deprecated_feature(LOG, msg)
+
+    return driver
 
 
 class _TraceMeta(type):
